docs/tasks.md

```diff
@@ -394,6 +405,8 @@ To access an input parameter, replace `resources` with `params`.
 ${inputs.params.<name>}
 ```
 
+Note that for parameters of type `array`, each element of the array will be appended together separated by spaces when used in templating.
```

Comment:
Is this still true?  Bit confused between this and the paragraph in your PR description:

> The implementation is defined such that an Array parameter can only be used if the replacement string is completely isolated and within a field that is an array of strings (currently eligible fields are command and args). For instance, the webhook will prevent an array parameter from being used in the image field. Similarly, an array parameter used in a composite string, like "value=${array}", is invalid and will be caught. The decision was made to completely prevent any use of array parameters as strings because it clutters Tekton with unnecessary functionality and is naturally extensible beyond the intended scope. For instance, if the behavior was defined such that array parameters used in the context of strings were automatically converted by separating them with a space, then that relatively arbitrary implementation could be questioned and eventually lead to needing to specify a custom delimiter (like commas), and so on.

I think I'm misunderstanding - arrays might still be concatenated together with spaces?